os/arch/arm/src/amebad/amebad_flash.c

```diff
@@ -119,22 +119,15 @@ static ssize_t amebad_erase_page(size_t page)
 {
     flash_t flash;
     uint32_t address;
-    irqstate_t irqs;
-
     if (page > (AMEBAD_START_SECOTR + AMEBAD_NSECTORS)) {
         printf("Invalid page number\n");
         return -EFAULT;
     }
-    /* Disable IRQs while erasing sector */
-    irqs = irqsave();

```

Comment:
Is it okay there will be interrupt when erasing or writing to flash?
And I think this does not have big impact on improving performance..

Reply:
The **flash_write_lock** function is used to lock CPU when write or erase flash in flash_api.c.
I have test it again, seems adding irq_save does not have significant effect on the speed.
In this case, I will add irq_save back to be safe. Thanks!

Reply:
Okay. If there is no big difference, I think adding irq_save is better :)